client-app/pages/product.vue

```diff
@@ -409,9 +425,10 @@ watchEffect(() => {
   }
 });
 
-watchEffect(() => {
+watchEffect(async () => {
   if (productReviewsEnabled.value) {
     void fetchCustomerReviews(productReviewsPayload.value);
+    leaveFeedbackVisible.value = await canLeaveFeedback(productId.value, "Product");
```

Comment:
can we move it from watch? to get rid of api call in watch?  productReviewsEnabled is not reactive in the core.